Add theme values for offline

puneet-pdx opened the pull request:
<!-- PRs should provide sufficient context on the changes, either by linking to the associated issue and/or by providing a summary. Also provide a link to the design if it's appropriate. -->

Related to issue: # https://devtopia.esri.com/runtime/kotlin/issues/5919

<!-- link to design, if applicable -->

### Description:

Add color and theme components used in the composable as a theme property so user can modify them.

### Summary of changes:

- Add `Theme` where the typography and colors are specified
- Add doc 

### Pre-merge Checklist

<!-- Tick one box for each section -->
- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/846/
  - [x] https://runtime-kotlin.esri.com/view/Release/job/200.8.0/job/vtest/view/vTest/job/toolkit/3/
  - [x] https://runtime-kotlin.esri.com/view/Release/job/200.8.0/job/vtest/view/vTest/job/toolkit/4/
  - [x] https://runtime-kotlin.esri.com/view/Release/job/200.8.0/job/vtest/view/vTest/job/toolkit/5/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

shubham7109 commented on toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/theme/Theme.kt:
```diff
@@ -0,0 +1,200 @@
+        cancelDownloadButtonColor: Color = ButtonDefaults.buttonColors().containerColor,
+        onDemandMapAreaSelectorAreaNameDialogBackgroundColor: Color = MaterialTheme.colorScheme.background
+    ): ColorScheme = ColorScheme(
+        downloadButtonColor = downloadButtonColor,
+        cancelButtonColor = cancelButtonColor,
+        cancelDownloadButtonColor = cancelDownloadButtonColor,
```
Should we consider this color as a `primaryButtonColor` which applies this color as the common icon-tint/button-color for:

- DownloadButton
- OpenButton
- AddMapAreaButton
- RefreshButton
- Rename map area button text
- OnDemandMapAreaSelectorOptions download button
- AreaNameDialog confirm button


And a `cancelButtonColor` which applies this color as the common icon-tint/button-color for:

- CancelButton
- CancelDownloadButtonWithProgressIndicator
- AreaNameDialog dismiss button
- FilledTonalIconButton close icon in OnDemandMapAreaSelectorOptions


Both these button colors could use the same default: `MaterialTheme.colorScheme.primary`.

puneet-pdx replied:
I have updated the buttons to be of the the color `MaterialTheme.colorScheme.primary`

shubham7109 commented on toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/theme/Theme.kt:
```diff
@@ -0,0 +1,200 @@
+    )
+}
+
+
+
```
Remove extra lines in eof.

puneet-pdx replied:
updated

shubham7109 commented on toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/theme/Theme.kt:
```diff
@@ -0,0 +1,200 @@
+ * limitations under the License.
+ */
+
+package com.arcgismaps.toolkit.offline.theme
```
Could you rename this file to be `OfflineMapAreasTheme.kt`

shubham7109 commented on toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/internal/utils/Buttons.kt:
```diff
@@ -79,29 +81,29 @@ internal fun CancelDownloadButtonWithProgressIndicator(progress: Int, onClick: (
                 .size(10.dp)
                 .clip(RectangleShape)
-                .background(ButtonDefaults.buttonColors().containerColor)
+                .background(colorScheme.cancelDownloadButtonColor)
```
Let's also set the `trackColor` of the `CircularProgressIndicator` of this `CancelDownloadButtonWithProgressIndicator` to use `colorScheme.cancelDownloadButtonColor`.

puneet-pdx replied:
The `CircularProgressIndicator` is tied to the theme's primary color.

puneet-pdx commented on toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/theme/Theme.kt:
```diff
@@ -0,0 +1,200 @@
+        downloadButtonColor = downloadButtonColor,
+        cancelButtonColor = cancelButtonColor,
+        cancelDownloadButtonColor = cancelDownloadButtonColor,
+        onDemandMapAreaSelectorAreaNameDialogBackgroundColor = onDemandMapAreaSelectorAreaNameDialogBackgroundColor
```
I have updated all of these to `MaterialTheme.colorScheme.background`

shubham7109 commented on toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/ondemand/OnDemandMapAreaSelector.kt:
```diff
@@ -169,27 +179,30 @@ private fun OnDemandMapAreaSelectorOptions(
-        modifier = Modifier.fillMaxWidth(),
+        modifier = Modifier
+            .fillMaxWidth()
+            .background(colorScheme.offlineBackgroundColor),
```
Similar comment here, the background could be applied in the top level `ModalBottomSheet` 's `containerColor`: 

```kotlin
ModalBottomSheet(
    modifier = Modifier.systemBarsPadding(),
    containerColor = colorScheme.offlineBackgroundColor,
....
```


```suggestion
```

shubham7109 commented on toolkit/offline/src/main/java/com/arcgismaps/toolkit/offline/ui/MapAreaDetailsScreen.kt:
```diff
@@ -108,12 +114,15 @@ internal fun MapAreaDetailsScreen(
     val scrollState = rememberScrollState()
     Column(
         modifier = Modifier
+            .background(colorScheme.offlineBackgroundColor)
```
Similar comment here, the background could be applied in the top level `ModalBottomSheet` 's `containerColor`: 

```kotlin
ModalBottomSheet(
  containerColor = colorScheme.offlineBackgroundColor,
....
```


```suggestion
```

shubham7109 reviewed (commented):
Thanks for adding in the feedback and changes are looking nice from my testing! The defaults used looks good, and I like the choices for color & typography provided to override the defaults. 

Added few comments to set the background color from a top-down approach:

puneet-pdx commented:
@shubham7109 I have addressed your comments. I have also pushed a fix for a build warning we are getting for the module. I thought it would be easier to just get it in here in this PR.
